### PR TITLE
System.Net EventSource Logging

### PR DESCRIPTION
**Pull request by josguil**
Adding EventSources to substitute the previous logging model using
TraceSources.

In this PR 4 event sources are added:

-NetEventSource
-SocketsEventSource
-HttpEventSource
-SecurityEventSource

Event methods for NetEventSource, SocketsEventSource and HttpEventSource
are fully added. SecurityEventSource has only a few of
the new event methods implemented.
Changes in production code are done to match this new implementation.
(Only changes for events in Sockets, Http are fully done in this iteration).

### CODE REVIEW

**Comment by josguil** [comment]
@CIPop @davidsh @vancem @stephentoub @pgavlin @weshaggard @ericstj PTAL.

File: src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
     <RootNamespace>System.Net.Http.Unit.Tests</RootNamespace>
     <AssemblyName>System.Net.Http.Unit.Tests</AssemblyName>
     <StringResourcesPath>..\..\src\Resources\Strings.resx</StringResourcesPath>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (davidsh) [inline src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj]: Why are we adding 'unsafe' code to Http?

File: src/System.Net.Security/src/Resources/Strings.resx [hunk 2]
```diff
@@ -378,4 +378,4 @@
   <data name="net_ssl_invalid_certificate" xml:space="preserve">
     <value>SSL certificate returned is invalid, OpenSSL error - {0}.</value>
   </data>
-</root>
+</root>
```
Comment (davidsh) [inline src/System.Net.Security/src/Resources/Strings.resx]: Add newline to end of file.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        LocalizationResources = "FxResources.System.Net.Http.SR")]
+    internal sealed class HttpEventSource : EventSource
+    {
+        private const int ASSOCIATE_ID = 1;
+        private const int URI_BASE_ADDRESS_ID = 2;
+        private const int CONTENT_NULL_ID = 3;
+        private const int CLIENT_SEND_COMPLETED = 4;
+        private const int HEADERS_INVALID_VALUE_ID = 5;
+
+        private static HttpEventSource s_log = new HttpEventSource();
```
Comment (justinvp) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: `readonly`

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[2].Size = (objectB.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&objectBHash);
+                dataDesc[3].Size = sizeof(int);
+
+                WriteEventCore(ASSOCIATE_ID, SIZEDATA, dataDesc);
+            }
+        }
+        [NonEvent]
```
Comment (justinvp) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: Should there be a blank line between methods? (Applies throughout)

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+    [EventSource(Name = "Microsoft-System-Net",
+        Guid = "501a994a-eb63-5415-9af1-1b031260f16c")]
+    internal sealed class NetEventSource : EventSource
+    {
+        private const int FUNCTIONSTART_ID = 1;
+        private const int FUNCTIONSTOP_ID = 2;
+        private const int CRITICALEXCEPTION_ID = 3;
+        private const int CRITICALERROR_ID = 4;
+
+        private static NetEventSource s_log = new NetEventSource();
```
Comment (justinvp) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: `readonly`

File: src/Common/src/System/Net/Logging/SecurityEventSource.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+namespace System.Net
+{
+    [EventSource(Name = "Microsoft-System-Net-Security",
+        Guid = "066c0e27-a02d-5a98-9a4d-078cc3b1a896",
+        LocalizationResources = "FxResources.System.Net.Security.SR")]
+    internal sealed class SecurityEventSource : EventSource
+    {
+        private const int ENUMERATE_SECURITY_PACKAGES_ID = 1;
+
+        private static SecurityEventSource s_log = new SecurityEventSource();
```
Comment (justinvp) [inline src/Common/src/System/Net/Logging/SecurityEventSource.cs]: `readonly`

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        private const int ACCEPTED_ID = 1;
+        private const int CONNECTED_ID = 2;
+        private const int CONNECTED_ASYNC_DNS_ID = 3;
+        private const int NOT_LOGGED_FILE_ID = 4;
+        private const int DUMP_ARRAY_ID = 5;
+        private const int DUMP_ARRAY_ASYNC_ID = 6;
+
+        private const int DefaultMaxDumpSize = 1024;
+
+        private static SocketsEventSource s_log = new SocketsEventSource();
```
Comment (justinvp) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: `readonly`

File: src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj (hunk 1 above)
Comment (josguil) [inline src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj]: Some of the optimized methods for logging need to use pointers.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+                            LoggingHash.HashInt(objA),
+                            LoggingHash.GetObjectName(objB),
+                            LoggingHash.HashInt(objB));
+        }
+
+        [Event(ASSOCIATE_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational, Message = "[{0}#{1}]<-->[{2}#{3}]")]
+        internal unsafe void Associate(string objectA, int objectAHash, string objectB, int objectBHash)
+        {
+            const int SIZEDATA = 4;
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: Naming convention: `sizeData`. (I think you can run the formatting tool to automatically correct these.)

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+using System.Globalization;
+
+namespace System.Net
+{
+    [EventSource(Name = "Microsoft-System-Net-Http",
+        Guid = "bdd9a83e-1929-5482-0d73-2fe5e1c0e16d",
+        LocalizationResources = "FxResources.System.Net.Http.SR")]
+    internal sealed class HttpEventSource : EventSource
+    {
+        private const int ASSOCIATE_ID = 1;
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: Naming convention: AssociateId, etc (you can run the formatting tool to correct these).

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+            Level = EventLevel.Informational)]
+        internal unsafe void UriBaseAddress(string uriBaseAddress, string objName, int objHash)
+        {
+            const int SIZEDATA = 3;
+            fixed (char* arg1Ptr = uriBaseAddress, arg2Ptr = objName)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+
+                dataDesc[0].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[0].Size = (uriBaseAddress.Length + 1) * sizeof(char); // Size in bytes, including a null terminator.
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Are you sure there _is_ a null terminator?

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+        private HttpEventSource() { }
+        public static HttpEventSource Log
+        {
+            get
+            {
+                return s_log;
+            }
+        }
+
+        internal static void Associate(object objA, object objB)
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Is this supposed to have a `[NonEvent]` attribute?

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+            }
+        }
+
+        [NonEvent]
+        internal static void ContentNull(object obj)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: New line after closing `}` if followed by a statement.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+        }
+
+        [NonEvent]
+        internal static void ClientSendCompleted(HttpClient httpClient, HttpResponseMessage response, HttpRequestMessage request)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            string responseString = "";
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: New line after closing } if followed by a statement.
Throughout the files (will stop adding comments for this issue).

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+            if (response != null)
+            {
+                responseString = response.ToString();
+            }
+            s_log.ClientSendCompleted(LoggingHash.HashInt(request), LoggingHash.HashInt(response), responseString, LoggingHash.HashInt(httpClient));
+        }
+
+        [Event(CLIENT_SEND_COMPLETED, Keywords = Keywords.Debug,
+            Level = EventLevel.Verbose)]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Nit: The Event args are split on two separate lines but the function below is not.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+        [Event(HEADERS_INVALID_VALUE_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Verbose)]
+        internal void HeadersInvalidValue(string name, string rawValue)
+        {
+            WriteEvent(HEADERS_INVALID_VALUE_ID, name, rawValue);
+        }
+
+        public class Keywords
+        {
+            public const EventKeywords Default = (EventKeywords)0x0001;
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: What convention did you use to split between Default and Debug? (throughout the implementation)

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 16]
```diff
@@ -0,0 +1,266 @@
...
+            s_log.FunctionStart(callerName,
+                                callerHash,
+                                LoggingHash.GetObjectName(method),
+                                parametersName,
+                                parametersHash,
+                                componentType);
+        }
+
+        [Event(FUNCTIONSTART_ID, Keywords = Keywords.FunctionEntryExit,
+    Level = EventLevel.Verbose, Message = "[{5}] {0}#{1}::{2}({3}#{4})")]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: Nit: alignment. (Also, our standard is to separate each argument on a new line.)

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+        internal unsafe void FunctionStart(string callerName,
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: Nit: if you need to separate arguments for a function, please start with the first argument.
E.g. 

```C#
internal unsafe void FunctionStart(
                               string callerName,
```

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 18]
```diff
@@ -0,0 +1,266 @@
...
+            s_log.FunctionStop(callerName,
+                    callerHash,
+                    LoggingHash.GetObjectName(method),
+                    parametersName,
+                    parametersHash,
+                    componentType);
+        }
+
+        [Event(FUNCTIONSTOP_ID, Keywords = Keywords.FunctionEntryExit,
+    Level = EventLevel.Verbose, Message = "[{5}] {0}#{1}::{2}({3}#{4})")]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: nit: alignment

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 19]
```diff
@@ -0,0 +1,266 @@
...
+        {
+            s_log.CriticalError(LoggingHash.GetObjectName(msg),
+                                LoggingHash.GetObjectName(method),
+                                LoggingHash.GetObjectName(obj),
+                                LoggingHash.HashInt(obj),
+                                componentType);
+        }
+
+        [Event(CRITICALERROR_ID, Keywords = Keywords.Default,
+Level = EventLevel.Critical)]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: nit: alignment

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 20]
```diff
@@ -0,0 +1,266 @@
...
+                dataDesc[4].DataPointer = (IntPtr)(&componentType);
+                dataDesc[4].Size = sizeof(int);
+                WriteEventCore(CRITICALERROR_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        public class Keywords
+        {
+            public const EventKeywords Default = (EventKeywords)0x0001;
+            public const EventKeywords Debug = (EventKeywords)0x0002;
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: Can we extract this class and reuse it instead of defining it for each eventsource?
I noticed that `Debug` isn't used in this event-source but it's still defined (to keep numbering).

File: src/Common/src/System/Net/Logging/NetEventSource.cs [hunk 21]
```diff
@@ -0,0 +1,266 @@
...
+            public const EventKeywords FunctionEntryExit = (EventKeywords)0x0004;
+        }
+
+        public enum ComponentType
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: Should this be in a separate file or is it only used in association with this class?

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 22]
```diff
@@ -0,0 +1,268 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace System.Net
+{
+    //TODO: If localization resources are not found, logging does not work.
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Please add more context why this is relevant here (this EventSource is included in multiple projects? Which ones?).
Also, please add an issue reference.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 9 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: I was following the same format this other event source class was using: https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/Threading/Tasks/TPLETWProvider.cs 

But yeah, I'm reading our coding style and PascalCasing should be used.

File: src/Common/src/System/Net/Logging/SecurityEventSource.cs [hunk 23]
```diff
@@ -0,0 +1,557 @@
...
+        private const int SELECTED_CERT_ID = 22;
+        private const int CERTS_AFTER_FILTERING_ID = 23;
+        private const int FINDING_MATCHING_CERTS_ID = 24;
+        private const int USING_CACHED_CREDENTIAL_ID = 25;
+        private const int SSPI_SELECTED_CIPHER_SUITE_ID = 26;
+        private const int REMOTE_CERTIFICATE_ERROR_ID = 27;
+        private const int REMOTE_CERTIFICATE_VALID_ID = 28;
+        private const int REMOTE_CERTIFICATE_SUCCESS_ID = 29;
+        private const int REMOTE_CERTIFICATE_INVALID_ID = 30;
+
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SecurityEventSource.cs]: Nit: extra new line here.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 10 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: No, but this is the pattern suggested in the eventsource guide, and is also the way the predefined WriteCore methods work. 
https://github.com/dotnet/coreclr/blob/ed1867bc8e445806ebe2e05a5d3515787006321d/src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 11 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Yes!

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 15 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: GlobalLog for some of the events. For the main majority of events, I used debug since it felt that default should only be used for a few default events. But I don't have any strong argument to say a specific event should have one keyword or another. We could go through the list of events and decide if the keyword should change for some events.

File: src/Common/src/System/Net/Logging/NetEventSource.cs (hunk 20 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: It doesn't work. From Vance's blog (http://blogs.msdn.com/b/vancem/archive/2012/08/14/etw-in-c-controlling-which-events-get-logged-in-an-system-diagnostics-tracing-eventsource.aspx):

"In addition to the event definitions themselves, there is also a nested class called 'Keywords' (you MUST use this name and it MUST be nested) which defined the numeric values for the keywords used in the event definitions.   Each of these should be a binary bit in the 64 bit long value.  "

File: src/Common/src/System/Net/Logging/NetEventSource.cs (hunk 21 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: It's only used in association with this class. No other event source needs this enum.

File: src/Common/src/System/Net/Logging/SecurityEventSource.cs [hunk 24]
```diff
@@ -0,0 +1,557 @@
...
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            WriteEvent(LOCATING_PRIVATE_KEY_ID, x509Certificate, secureChannelHash);
+        }
+
+        [Event(CERT_IS_TYPE_2_ID, Keywords = Keywords.Default,
+    Level = EventLevel.Informational)]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SecurityEventSource.cs]: nit: alignment

File: src/Common/src/System/Net/Logging/SecurityEventSource.cs [hunk 25]
```diff
@@ -0,0 +1,557 @@
...
+                dataDesc[2].DataPointer = (IntPtr)(&inFlags);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(ACCEPT_SECURITY_CONTEXT_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [Event(OPERATION_RETURNED_SOMETHING_ID, Keywords = Keywords.Default,
+            Level = EventLevel.Informational)]
+        internal void OperationReturnedSomething(string operation, Interop.SecurityStatus errorCode)
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SecurityEventSource.cs]: This needs to be renamed: OperationReturned ?

File: src/Common/src/System/Net/Logging/SecurityEventSource.cs [hunk 26]
```diff
@@ -0,0 +1,557 @@
...
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            WriteEvent(NO_ISSUERS_TRY_ALL_CERTS_ID, secureChannelHash);
+        }
+
+        [Event(LOOK_FOR_MATCHING_CERTS_ID, Keywords = Keywords.Default,
+Level = EventLevel.Informational)]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SecurityEventSource.cs]: nit: allignment

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 27]
```diff
@@ -0,0 +1,268 @@
...
+            if ((length < 0) || (length > buffer.Length - offset))
+            {
+                length = buffer.Length - offset;
+            }
+
+            byte[] partialBuffer = new byte[length];
+
+            for (int i = 0; i < length; i++)
+            {
+                partialBuffer[i] = buffer[i + offset];
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Please use Buffer.BlockCopy instead of iterating through an array.

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 28]
```diff
@@ -0,0 +1,268 @@
...
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            if (length > DefaultMaxDumpSize)
+            {
+                length = DefaultMaxDumpSize;
+            }
+            byte[] buffer = new byte[length];
+            Marshal.Copy(bufferPtr, buffer, 0, length);
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Why does this need to copy the data but the other events are only using pointers to existing objects?

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 29]
```diff
@@ -0,0 +1,268 @@
...
+            const int SIZEDATA = 3;
+            EventSource.EventData* descrs = stackalloc EventSource.EventData[SIZEDATA];
+            descrs[0].DataPointer = (IntPtr)(&method);
+            descrs[0].Size = sizeof(int);
+            if (buffer == null || buffer.Length == 0)
+            {
+                int blobSize = 0;
+                descrs[1].DataPointer = (IntPtr)(&blobSize);
+                descrs[1].Size = 4;
+                descrs[2].DataPointer = (IntPtr)(&blobSize); // valid address instead of empty contents
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Why is this needed?

Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Nit: Comments should start with capital and end with "."

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 30]
```diff
@@ -0,0 +1,268 @@
...
+                    descrs[1].Size = 4;
+                    descrs[2].DataPointer = (IntPtr)blob;
+                    descrs[2].Size = blobSize;
+                    WriteEventCore(DUMP_ARRAY_ID, SIZEDATA, descrs);
+                }
+            }
+        }
+
+        [Event(DUMP_ARRAY_ASYNC_ID, Keywords = Keywords.Default,
+    Level = EventLevel.Informational, Message = "FinishOperation({0}Async)")]
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Why does this have a message that is not localized?

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 31]
```diff
@@ -0,0 +1,268 @@
...
+                length = DefaultMaxDumpSize;
+            }
+            byte[] buffer = new byte[length];
+            Marshal.Copy(bufferPtr, buffer, 0, length);
+
+            if (method is MethodType)
+            {
+                s_log.DebugDumpArray((MethodType)method, buffer);
+            }
+            else if (method is SocketAsyncOperation)
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Why would it matter where the byte dump is made? We have Enter/Leave for that.

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs [hunk 32]
```diff
@@ -0,0 +1,268 @@
...
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(&socketHash);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(NOT_LOGGED_FILE_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void Dump(object method, IntPtr bufferPtr, int length)
```
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: I don't think we need to have the `method` parameter. We have enough information without it and it's adding a lot of complication that may not work on NetNative.

File: src/System.Net.Http/src/Resources/Strings.resx [hunk 33]
```diff
@@ -354,4 +354,4 @@
   <data name="net_http_unix_invalid_response" xml:space="preserve">
     <value>The server returned an invalid or unrecognized response.</value>
   </data>
-</root>
+</root>
```
Comment (CIPop) [inline src/System.Net.Http/src/Resources/Strings.resx]: Please undo this change: we do like a new line at EOF.

File: src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj (hunk 1 above)
Comment (CIPop) [inline src/System.Net.Http/tests/UnitTests/System.Net.Http.Unit.Tests.csproj]: But this is the Unit Test code: I assume this is because it's using actual ETW logging instead of creating a Fake log?

File: src/System.Net.Security/src/Resources/Strings.resx (hunk 2 above)
Comment (CIPop) [inline src/System.Net.Security/src/Resources/Strings.resx]: +1

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 10 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: I can't find `WriteCore` in that file.
If there is no null terminator then this is an off-by-one buffer overrun which is a security bug. 

Please investigate and, if that is not a terminator,  open GitHub issues for other places where you've observed the incorrect pattern.

File: src/Common/src/System/Net/Logging/NetEventSource.cs (hunk 20 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/NetEventSource.cs]: Good to know. Thanks!

**Comment by CIPop** [comment]
/cc: @vancem There's one thing that I'd like clarified before merging: the potential buffer overrun in the pattern used to call the logging functions (should be simple to take a look with a debugger). 
According to this: http://stackoverflow.com/questions/6293457/why-are-c-net-strings-length-prefixed-and-null-terminated it should work as currently written.

LGTM otherwise but cannot be merged until the checks pass. 

@josguil The errors are:

> `error CS0117: 'SR' does not contain a definition for 'net_log_remote_certificate' [d:\j\workspace\dotnet_corefx\ubuntu_debug_bld_prtest\src\System.Net.Security\src\System.Net.Security.csproj]
Indication 1`

These could be caused by missing some of the logging functions in the PAL layer code?

File: src/System.Net.Http/src/Resources/Strings.resx (hunk 33 above)
Comment (josguil) [inline src/System.Net.Http/src/Resources/Strings.resx]: I will. When I modify the file in VS it does that change automatically.

**Comment by josguil** [comment]
In my last commit I also edited a logging line in the PAL layer. It should solve that SR error.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 10 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: My bad, I meant WriteEvent.

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs (hunk 31 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Some of these Dumps happen in Async operations, so there may be different dumps having similar time stamps and therefore making it difficult to decide to which Enter/Exit method it is related.

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs (hunk 32 above)
Comment (josguil) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: Is the complication the "if (method is MethodType)" check? I could separate the dump method into two to avoid that check. I'm not totally convinced we don't need that parameter. It will make easier to know where the dump is coming from even if that information can be inferred from the NetEventSource Enter and Exit methods. And since it's just an extra int, compared to the whole buffer we're logging, it adds very little overhead.

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 10 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: Saw it now: https://github.com/dotnet/coreclr/blob/ed1867bc8e445806ebe2e05a5d3515787006321d/src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs#L920

File: src/Common/src/System/Net/Logging/HttpEventSource.cs (hunk 15 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/HttpEventSource.cs]: I don't understand this part:

> GlobalLog for some of the events. 

Was that set to Debug?

> But I don't have any strong argument to say a specific event should have one keyword or another.

We should think of a clear statement on what is Default vs Debug. The least functional requirement should be that when only Default is used, the trace should still be useful.

**Comment by josguil** [comment]
@dotnet-bot test this please

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs (hunk 31 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: I don't think this is very useful: the socket object should be the one that makes the link, not the type. I would expect that the majority of apps would either use Async or Sync but never combine them.

File: src/Common/src/System/Net/Logging/SocketsEventSource.cs (hunk 32 above)
Comment (CIPop) [inline src/Common/src/System/Net/Logging/SocketsEventSource.cs]: See my comment above.

**Comment by josguil** [comment]
Thanks for the review. 

List of tracking issues: 

https://github.com/dotnet/corefx/issues/5144
